app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsViewModel.kt

```diff
@@ -41,7 +43,7 @@ class TransactionDetailsViewModel
             executionInfo.signers.contains(credentials.address) && !executionInfo.confirmations.map { it.signer }.contains(credentials.address)
         }
 
-    fun submitConfirmation(transaction: TransactionDetails, executionInfo: DetailedExecutionInfo.MultisigExecutionDetails) {
+    fun submitConfirmation(transaction: TransactionDetailsViewData, executionInfo: DetailedExecutionInfo.MultisigExecutionDetails) {
```

Comment:
```suggestion
   var txDetails: TransactionDetails
   private set

    fun submitConfirmation(transaction: TransactionDetails) {
      val executionInfo = kotlin.runCatching { 
         txDetails.detailedExecutionInfo as DetailedExecutionInfo.MultisigExecutionDetails 
         }.onSuccess {
         it
         }.onFailure {
         throw MissingCorrectExecutionDetailsException
         }
```

Reply:
This looks incomplete. Is there a way to see the complete suggested change?